src/test/java/com/cloudant/tests/ResponseTest.java

```diff
@@ -0,0 +1,86 @@
+package com.cloudant.tests;
+
+import static org.junit.Assert.assertEquals;
+
+import com.cloudant.client.api.model.Response;
+import com.cloudant.client.org.lightcouch.CouchDbException;
+import com.cloudant.client.org.lightcouch.DocumentConflictException;
+import com.cloudant.client.org.lightcouch.NoDocumentException;
+import com.cloudant.tests.util.CloudantClientResource;
+import com.cloudant.tests.util.DatabaseResource;
+import com.cloudant.tests.util.TestLog;
+
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Test cases to verify status code from Response object.
+ * Assert status codes in CouchDbException and its subclasses.
+ */
+public class ResponseTest {
+    @ClassRule
+    public static final TestLog TEST_LOG = new TestLog();
+
+    @ClassRule
+    public static CloudantClientResource clientResource = new CloudantClientResource();
+    @Rule
```

Comment:
Creating a new database for each test method can make the tests quite a lot slower.
Can we just use the `ChainRule` for a single database for the duration of the testing and just using different document IDs?

```java
private Foo foo;
@Before
public void newFoo(){
foo = new Foo(Utils.generateUUID())
}
```